compiler/src/dmd/escape.d

```diff
@@ -781,7 +781,8 @@ bool checkAssignEscape(ref Scope sc, Expression e, bool gag, bool byRef)
              * It may escape via that assignment, therefore, v can never be 'scope'.
              */
             //printf("no infer for %s in %s, %d\n", v.toChars(), fd.ident.toChars(), __LINE__);
-            doNotInferScope(v, e);
+            if (!v.isParameter)
+                notMaybeScope(v, e);
         }
```

Comment:
`doNotInferScope` at least says something that can be understood. `notMaybeScope` is a terrible name